src/main/java/nextstep/qna/domain/Answers.java

```diff
@@ -0,0 +1,38 @@
+package nextstep.qna.domain;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import nextstep.qna.CannotDeleteException;
+import nextstep.users.domain.NsUser;
+
+public class Answers {
+
+    private final List<Answer> answers;
+
+    public Answers(List<Answer> answers) {
+        this.answers = answers;
+    }
+
+    public void add(Answer answer) {
+        answers.add(answer);
+    }
+
+    public void delete(List<DeleteHistory> deleteHistories, NsUser user) throws CannotDeleteException {
+        canDelete(user);
+        for (Answer answer : answers) {
+            answer.setDeleted(true);
```

Comment:
유효성을 검증하고 setter를 호출하기 보다 메시지를 보내 검증하면 어떨까요?
```suggestion
        for (Answer answer : answers) {
            answer.delete(user);
```